File: client/src/app/modules/miscellaneous/services/document.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DocumentType } from '../../miscellaneous/model/DocumentType';

@Injectable({
  providedIn: 'root'
})
export class DocumentTypeService extends AppRestService<DocumentType>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getDocumentTypes() {
    return this.getListCached(new HttpParams(), "document-types");
  }

  addOrUpdateDocumentType(documentType: DocumentType) {
    this.clearListCache(new HttpParams(), "document-types");
    return this.addOrUpdate(new HttpParams(), "document-type", documentType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/DocumentTypeService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.DocumentType;

public interface DocumentTypeService {
    public List<DocumentType> getDocumentTypes();

    public DocumentType getDocumentType(Integer id);

    public DocumentType addOrUpdateDocumentType(DocumentType documentType);
}
